pkg/services/ldap/ldap.go

```diff
@@ -192,12 +202,40 @@ func (auth *Auth) SyncUser(query *models.LoginUserQuery) error {
 	return nil
 }
 
+// DisableExternalUser marks external user as disabled in Grafana db
+func (auth *Auth) DisableExternalUser(username string) error {
+	// Check if external user exist in Grafana
+	userQuery := &models.GetExternalUserInfoByLoginQuery{
+		LoginOrEmail: username,
+	}
+
+	if err := bus.Dispatch(userQuery); err != nil {
+		return err
+	}
+
+	userInfo := userQuery.Result
+	if userInfo.AuthModule == models.AuthModuleLDAP && !userInfo.IsDisabled {
```

Comment:
We should be able to skip the `userInfo.AuthModule == models.AuthModuleLDAP` check here. This should only be called when a user hasnt been found by ldap. Right?

Reply:
Yes, it's only for the users not found in LDAP.

Reply:
Then we should remove it, simplify the code a bit.